package.json

```diff
@@ -468,7 +477,16 @@
           }
         ]
       }
-    ]
+    ],
+    "menus": {
+      "debug/toolBar": [
+        {
+          "command": "cdt.debug.resumeAllSession",
+          "group": "navigation",
+          "when": "debugType == cdt-amalgamator && debugState == stopped"
```

Comment:
> 1- `debugType == cdt-amalgamator` will work during dev, but the button won't appear for the vendor specific amalgamator types (E.g. renesas-amalgamator). It seems to me that some other context state should be used to determine if the target supports resume all and make it conditional on that

I think will work in case using directly cdt-amalgamator to debug. Do you have any suggestion for this case?

Reply:
> 2- Should this really be limited to `debugState == stopped`? What if the current thread is running and I want to resume all the threads, will this mean that I have to change threads to one of the stopped threads before I can resume all?

The condition is true when any session meets, don't need to all sessions have to stop state. You can check bellow images
![MicrosoftTeams-image (36)](https://user-images.githubusercontent.com/104537989/217810266-d9726790-7055-4b05-869a-26d3aa4c6ef3.png)

![MicrosoftTeams-image (37)](https://user-images.githubusercontent.com/104537989/217810553-8a3f79e9-400b-49db-a7d1-17c86638e83f.png)

Reply:
> > 1- `debugType == cdt-amalgamator` will work during dev, but the button won't appear for the vendor specific amalgamator types (E.g. renesas-amalgamator). It seems to me that some other context state should be used to determine if the target supports resume all and make it conditional on that
> 
> I think will work in case using directly cdt-amalgamator to debug. Do you have any suggestion for this case?

Is the type `cdt-amalgamator`? The package.json seems to have it as `amalgamator`:

https://github.com/eclipse-cdt-cloud/cdt-amalgamator/blob/03fabfeb80134c5820387af79fcc54eb0fbf11ca/package.json#L61-L63

Reply:
> ![MicrosoftTeams-image (36)](https://user-images.githubusercontent.com/104537989/217810266-d9726790-7055-4b05-869a-26d3aa4c6ef3.png)

Your selection in the above picture is the stopped thread. My question was about when the current (aka selected) thread is running. Here is a screenshot with running thread selected

![image](https://user-images.githubusercontent.com/679236/217982519-27d9881c-d385-46d1-b007-af82cfcb9dd7.png)

So, back to my original question (this is a question, not necessarily something that you need to change unless you realize there is a mistake now that you are reexamining this question):

2- Should this really be limited to debugState == stopped? What if the current thread is running and I want to resume all the threads, will this mean that I have to change threads to one of the stopped threads before I can resume all?

Reply:
> 2- Should this really be limited to debugState == stopped? What if the current thread is running and I want to resume all the threads, will this mean that I have to change threads to one of the stopped threads before I can resume all?

I have considered and removed the condition "debugState == stopped". With the current thread running, there is no need to change its state to "stopped" prior to pressing the "resume all" button. I think it will be more convenient for users.

Reply:
> > > 1- `debugType == cdt-amalgamator` will work during dev, but the button won't appear for the vendor specific amalgamator types (E.g. renesas-amalgamator). It seems to me that some other context state should be used to determine if the target supports resume all and make it conditional on that
> > 
> > 
> > I think will work in case using directly cdt-amalgamator to debug. Do you have any suggestion for this case?
> 
> Is the type `cdt-amalgamator`? The package.json seems to have it as `amalgamator`:
> 
> https://github.com/eclipse-cdt-cloud/cdt-amalgamator/blob/03fabfeb80134c5820387af79fcc54eb0fbf11ca/package.json#L61-L63

I updated it